drivers: spi: Initialize CS GPIOs in spi-bitbang

andysan opened the pull request:
The SPI bitbang driver doesn't correctly initialize the list of CS GPIOs. As a consequence, SPI buses using the bitbang driver won't drive CS low. Add the missing initialization.

Signed-off-by: Andreas Sandberg <andreas@sandberg.uk>

andysan committed:
drivers: spi: spi-bitbang: Initialize CS GPIOs

The SPI bitbang driver doesn't correctly initialize the list of CS
GPIOs. As a consequence, SPI buses using the bitbang driver won't
drive CS low. Add the missing initialization.

Signed-off-by: Andreas Sandberg <andreas@sandberg.uk>

bbilas commented:
+1 for this change but looking briefly at this driver, it seems that there is another issue.

`spi_context_cs_configure_all(&data->ctx);` shouldn't be called in the `spi_bitbang_configure` fuction but in the `spi_bitbang_init`. Would you mind fixing that as well?

andysan commented:
> +1 for this change but looking briefly at this driver, it seems that there is another issue.
> 
> 
> 
> `spi_context_cs_configure_all(&data->ctx);` shouldn't be called in the `spi_bitbang_configure` fuction but in the `spi_bitbang_init`. Would you mind fixing that as well?

I discovered that one as well yesterday and already have a patch. I'll post it tonight.

andysan committed:
drivers: spi: spi-bitbang: Fix CS glitch in transceive

spi_context_cs_configure_all() is currently called from
spi_bitbang_transceive(). This causes a glitch when combined with
SPI_HOLD_ON_CS is used.

Move the initialization to spi_bitbang_init which is what the other
SPI drivers seem to do.

Signed-off-by: Andreas Sandberg <andreas@sandberg.uk>
